src/components/ColorPicker.tsx

```diff
@@ -42,14 +49,74 @@ const ColorSwatch: FC<{
   const dispatch = useDispatch()
   const colors = useSelector(themeColors)
   const fontSize = useSelector(state => state.fontSize)
-  const selected =
-    (color && cursorStyle?.color === color) || (backgroundColor && cursorStyle?.backgroundColor === backgroundColor)
+  const currentThought = useSelector(state => !!state.cursor && getThoughtById(state, head(state.cursor))) as Thought
+  const simplePath = useSelector(state => simplifyPath(state, state.cursor!))
```

Comment:
`simplifyPath` does not return a stable reference, so this will cause the component to be re-rendered on every change to the Redux state. Set the equality function to https://github.com/cybersemics/em/blob/main/src/util/equalPath.ts to avoid this problem.

It is dangerous to use `!` here. Either handle the `null` case or add a comment that compellingly demonstrates that `state.cursor` could never be null at runtime.

Reply:
Since `simplePath` is only used when you dispatch `editThought`, I would recommend using a thunk rather than useSelector:

```ts
  dispatch((dispatch, getState) => {
    const state = getState()
    if (!state.cursor) return
    const thought = getThoughtById(state, head(state.cursor))
    const simplePath = simplifyPath(state, state.cursor)
    dispatch(editThought({ 
      oldValue: thought.value, 
      newValue: thoughtContentEditable.innerHTML, 
      path: simplePath 
    })),
  })
```